lib/internal/Magento/Framework/ObjectManager/Config/Compiled.php

```diff
@@ -120,6 +120,7 @@ public function getInstanceType($instanceName)
      */
     public function getPreference($type)
     {
+        $type = ltrim($type, '\\');
```

Comment:
This does not look like a part of in-object cache, which is the purpose of this PR